src/Symfony/Component/PasswordHasher/Tests/Hasher/NativePasswordHasherTest.php

```diff
@@ -103,12 +103,13 @@ public function testBcryptWithNulByte()
         $hasher = new NativePasswordHasher(null, null, 4, \PASSWORD_BCRYPT);
         $plainPassword = "a\0b";
 
-        if (\PHP_VERSION_ID < 80218 || \PHP_VERSION_ID >= 80300 && \PHP_VERSION_ID < 80305) {
-            // password_hash() does not accept passwords containing NUL bytes since PHP 8.2.18 and 8.3.5
-            $this->assertFalse($hasher->verify(password_hash($plainPassword, \PASSWORD_BCRYPT, ['cost' => 4]), $plainPassword));
-        }
+        try {
+            $verified = $hasher->verify($hasher->hash($plainPassword), $plainPassword);
 
-        $this->assertTrue($hasher->verify($hasher->hash($plainPassword), $plainPassword));
+            $this->assertTrue($verified);
+        } catch (\ValueError $error) {
+            $this->markTestSkipped(sprintf('password_hash() does not accept passwords containing NUL bytes on PHP %s. ', \PHP_VERSION));
```

Comment:
```suggestion
            $this->markTestSkipped('password_hash() does not accept passwords containing NUL bytes.');
```

I would not include the current PHP version here.